code/modules/mob/living/basic/boss/boss.dm

```diff
@@ -0,0 +1,165 @@
+	desc = "Attack the weak point for massive damage."
+	health = 1000
+	maxHealth = 1000
+	combat_mode = TRUE
+	sentience_type = SENTIENCE_BOSS
+	mob_biotypes = MOB_ORGANIC|MOB_SPECIAL
+	environment_smash = ENVIRONMENT_SMASH_RWALLS
+	damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 0, CLONE = 0, STAMINA = 0, OXY = 0)
+	unsuitable_cold_damage = 0
+	unsuitable_heat_damage = 0
+	unsuitable_atmos_damage = 0
+	move_force = MOVE_FORCE_OVERPOWERING
+	move_resist = MOVE_FORCE_OVERPOWERING
+	pull_force = MOVE_FORCE_OVERPOWERING
+	mob_size = MOB_SIZE_HUGE
+	layer = LARGE_MOB_LAYER
+	mouse_opacity = MOUSE_OPACITY_OPAQUE
+	flags_1 = PREVENT_CONTENTS_EXPLOSION_1
+	can_buckle_to = FALSE
+	faction = list(FACTION_MINING, FACTION_BOSS)
+	// Pale purple, should be red enough to see stuff on lavaland
+	lighting_cutoff_red = 25
+	lighting_cutoff_green = 15
+	lighting_cutoff_blue = 35
+	/// Loot dropped when the megafauna is NOT killed with a crusher
+	var/list/loot = list()
+	/// Crusher loot dropped when the megafauna is killed with a crusher
+	var/list/crusher_loot = list()
+	/// Achievement given to surrounding players when the megafauna is killed
+	var/achievement_type
+	/// Crusher achievement given to players when megafauna is killed
+	var/crusher_achievement_type
+	/// Score given to players when megafauna is killed
+	var/score_achievement_type
+	/// If the megafauna was actually killed (not just dying, then transforming into another type)
+	var/elimination = FALSE
+	/// Name for the GPS signal of the megafauna
+	var/gps_name = null
+	/// If this is a megafauna that is real (has achievements, gps signal)
+	var/true_spawn = TRUE
+	/// Attack actions, sets chosen_attack to the number in the action
+	var/list/attack_action_types = list()
+	/// Summoning line, said when summoned via megafauna vents.
+	var/summon_line = "I'll kick your ass!"
+	/// weather immunities
+	var/list/weather_immunities = list(TRAIT_LAVA_IMMUNE,TRAIT_ASHSTORM_IMMUNE)
+
+/mob/living/basic/boss/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/seethrough_mob)
+	AddElement(/datum/element/simple_flying)
+	if(gps_name && true_spawn)
+		AddComponent(/datum/component/gps, gps_name)
+	ADD_TRAIT(src, TRAIT_SPACEWALK, INNATE_TRAIT)
+	add_traits(list(TRAIT_NO_TELEPORT, TRAIT_MARTIAL_ARTS_IMMUNE, TRAIT_NO_FLOATING_ANIM), MEGAFAUNA_TRAIT)
+	add_traits(weather_immunities, ROUNDSTART_TRAIT)
+	grant_actions_by_list(attack_action_types)
+
+/mob/living/basic/boss/death(gibbed, list/force_grant)
+	if(gibbed) // in case they've been force dusted
+		return ..()
+
+	if(health > 0) // prevents instakills
+		return
+	var/datum/status_effect/crusher_damage/crusher_dmg = has_status_effect(/datum/status_effect/crusher_damage)
+	///Whether we killed the megafauna with primarily crusher damage or not
+	var/crusher_kill = (crusher_dmg && crusher_dmg.total_damage >= maxHealth * 0.6)
+	for(var/loot in (crusher_kill ? crusher_loot : loot)) // might aswell just do it here because the element proves unreliable
```

Comment:
Fix the element then, if it is broken we want it to work not to go back to recoding the same shit in multiple places.

Reply:
using two elements is not efficient and having to remove the other whenever someone gets a crusher kill is shit

Reply:
then we still want a general solution and not one specific to this typepath 🤷 
If you don't want to take the effort to make the code better you're not doing a refactor